server/src/main/java/io/spine/server/commandbus/CommandErrorHandler.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.commandbus;
+
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableSet;
+import io.spine.annotation.Internal;
+import io.spine.core.CommandClass;
+import io.spine.core.CommandEnvelope;
+import io.spine.core.Commands;
+import io.spine.core.Rejection;
+import io.spine.server.rejection.RejectionBus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.spine.core.Rejections.causedByRejection;
+import static java.lang.String.format;
+
+/**
+ * A {@link CommandDispatcher} which handles the dispatching errors.
+ *
+ * @author Dmytro Dashenkov
+ * @see #onError(CommandEnvelope, RuntimeException)
+ */
+@Internal
+public final class CommandErrorHandler<I> implements CommandDispatcher<I> {
```

Comment:
Let's discuss this in person. I don't see it like named like that. Given the responsibility it should have had — as we discussed it — it should have had a different name.